lib/mss/mss_utils.js

```diff
@@ -0,0 +1,800 @@
+        break;
+      case ContentType.AUDIO:
+        // moov/trak/mdia/minf/smhd
+        MssUtils.createSmhdBox_(isoBoxer, minf);
+        break;
+    }
+    // moov/trak/mdia/minf/dinf
+    const dinf = isoBoxer.createBox('dinf', minf);
+    // moov/trak/mdia/minf/dinf/dref
+    MssUtils.createDrefBox_(isoBoxer, dinf);
+    // moov/trak/mdia/minf/stbl
+    const stbl = isoBoxer.createBox('stbl', minf);
+    // Create empty stts, stsc, stco and stsz boxes
+    // Use data field as for codem-isoboxer unknown boxes for setting
+    // fields value
+    // moov/trak/mdia/minf/stbl/stts
+    const stts = isoBoxer.createFullBox('stts', stbl);
+    // version = 0, flags = 0, entry_count = 0
+    stts._data = [0, 0, 0, 0, 0, 0, 0, 0];
+    // moov/trak/mdia/minf/stbl/stsc
+    const stsc = isoBoxer.createFullBox('stsc', stbl);
+    // version = 0, flags = 0, entry_count = 0
+    stsc._data = [0, 0, 0, 0, 0, 0, 0, 0];
+    // moov/trak/mdia/minf/stbl/stco
+    const stco = isoBoxer.createFullBox('stco', stbl);
+    // version = 0, flags = 0, entry_count = 0
+    stco._data = [0, 0, 0, 0, 0, 0, 0, 0];
+    // moov/trak/mdia/minf/stbl/stsz
+    const stsz = isoBoxer.createFullBox('stsz', stbl);
+    // version = 0, flags = 0, sample_size = 0, sample_count = 0
+    stsz._data = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0];
+    // moov/trak/mdia/minf/stbl/stsd
+    MssUtils.createStsdBox_(isoBoxer, stbl, stream);
+    // moov/mvex
+    const mvex = isoBoxer.createBox('mvex', moov);
+    // moov/mvex/trex
+    MssUtils.createTrexBox_(isoBoxer, mvex, stream);
+    if (stream.encrypted) {
+      MssUtils.createProtectionSystemSpecificHeaderBox_(isoBoxer, moov, stream);
+    }
+  }
+
+  /**
+   * Create mvhd box.
+   *
+   * @param {ISOBoxer} isoBoxer
+   * @param {ISOBoxer} moov
+   * @param {shaka.extern.Stream} stream
+   * @private
+   */
+  static createMvhdBox_(isoBoxer, moov, stream) {
+    const mvhd = isoBoxer.createFullBox('mvhd', moov);
+    // version = 1  in order to have 64bits duration value
+    mvhd.version = 1;
+    // the creation time of the presentation => ignore (set to 0)
+    mvhd.creation_time = 0;
+    // the most recent time the presentation was modified => ignore (set to 0)
+    mvhd.modification_time = 0;
+    // the time-scale for the entire presentation => 10000000 for MSS
+    const timescale = stream.mssPrivateData.timescale;
+    mvhd.timescale = timescale;
+    // the length of the presentation (in the indicated timescale)
+    const duration = stream.mssPrivateData.duration;
+    mvhd.duration = duration === Infinity ?
+        0x1FFFFFFFFFFFFF : Math.round(duration * timescale);
+    // 16.16 number, '1.0' = normal playback
+    mvhd.rate = 1.0;
+    // 8.8 number, '1.0' = full volume
+    mvhd.volume = 1.0;
+    mvhd.reserved1 = 0;
+    mvhd.reserved2 = [0x0, 0x0];
+    mvhd.matrix = [
+      1, 0, 0, // provides a transformation matrix for the video;
+      0, 1, 0, // (u,v,w) are restricted here to (0,0,1)
+      0, 0, 16384,
```

Comment:
Good question.... I don't know, the code it based on https://github.com/Dash-Industry-Forum/dash.js/blob/development/src/mss/MssFragmentMoovProcessor.js#L165

Reply:
Oh, I think this might be a problem then.

Looking at that file, it looks like there's a high degree of similarity between these two files. I assume you mostly copied it, and then just re-formatted and tweaked some things for our specific use-case?

[Here's the license](https://github.com/Dash-Industry-Forum/dash.js/blob/development/LICENSE.md) that dash.js uses. It says:
> Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.

The license also says that the terms apply "with or without modification". So, while I am not a lawyer, I am fairly sure that would apply in this case.

For that reason, I don't think that putting the Shaka Player apache license header at the top of this file is appropriate. I think it MIGHT be enough to have a different license header. Like what we do in `docs/jsdoc-template/publish.js`. I'm not sure what it would look like though, so Joey's input will be needed.

Reply:
@theodab any update about it?